app/javascript/kramdown.jsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+
+/**
+ * Escapes user input for storage in a Kramdown extension attribute.
+ * Note that this _does not_ do sanitization--that'll be the job of
+ * whatever's rendering the Markdown to HTML.
+ * @param {any} input
+ * @returns {string}
+ */
+function escapeForExtensionAttribute(input) {
+  return String(input ?? "").replace(/"/g, '\\"');
+}
+
+/**
+ * @param {any} value
+ * @returns {string}
+ */
+function unescapeExtensionAttribute(value) {
+  return String(value ?? "").replace(/\\"/g, '"');
+}
+
+/**
+ * Generates a regular expression that will match a Kramdown extension, e.g.:
+ *
+ * ```
+ * {::some_tag attr="value" otherAttr="value" /}
+ * ```
+ *
+ * @param {name} {string}
+ * @param {attrNames} {string[]} Names of attributes expected (in order)
+ * @returns {RegExp} A regular expression that will match a kramdown extension tag.
+ */
+function makeExtensionRegex(name, attrNames) {
+  // NOTE: This pattern is quite fragile and depends on exact ordering of attributes.
+  const attrPatterns = attrNames
+    .map((attrName) => `${attrName}="(.*)"`)
+    .join(" ");
+  return new RegExp(`{::${name}${attrPatterns ? ` ${attrPatterns}` : ""} \\/}`);
+}
+
+/**
+ * Creates a Netlify Markdown editor component that is represented in markdown
+ * using Kramdown's extension syntax.
+ * @param options {Object}
+ * @param options.id {string}
+ * @param options.label {string}
+ * @param options.fields {{name: string, label: string, widget: string, [key: string]: any}[]} Fields supported by the component. (Order matters here)
```

Comment:
[question] is this the standard JS notation for an array of objects? I didn't realize it was an array until I went back to one of the files that calls this function

Reply:
Technically it's Typescript. I am only doing it here to get some autocomplete going in VSCode.